ex4/ex4.c

```diff
@@ -7,10 +7,29 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <sys/wait.h>
+#include <unistd.h>
+#include <string.h>
 
 int main(void)
 {
-    // Your code here    
-
+    int rc = fork();
+    
+    if (rc < 0) {
+        fprintf(stderr, "fork failed\n");
+        exit(1);
+    } else if (rc == 0) {
+        printf("hello, child here (pid: %d) \n", (int) getpid());
+        printf("hello\n");
+        char *myargs[3];
+        myargs[0] = strdup("wc");      // pass the name of the program we want to run as the first array element
```

Comment:
You are correct in saying that myargs[0] should be the program we want to run. Notice the instructions ask you to run  `/bin/ls`.